tests/unit/Diff/StatementListDifferPatcherTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Tests\Diff;
+
+use Wikibase\DataModel\Services\Diff\StatementListDiffer;
+use Wikibase\DataModel\Services\Diff\StatementListPatcher;
+use Wikibase\DataModel\Snak\PropertyNoValueSnak;
+use Wikibase\DataModel\Snak\SnakList;
+use Wikibase\DataModel\Statement\Statement;
+use Wikibase\DataModel\Statement\StatementList;
+
+class StatementListDifferPatcherTest extends \PHPUnit_Framework_TestCase {
+
+	public function testStatemetsAreMergedWhenPatched() {
+		$guid = 'some-guid';
+		$statementLatest = new Statement(
+			new PropertyNoValueSnak( 1 ),
+			new SnakList( [ new PropertyNoValueSnak( 3 ) ] ),
+			null,
+			$guid
+		);
+		$statementFrom = new Statement(
+			new PropertyNoValueSnak( 1 ),
+			new SnakList( [ new PropertyNoValueSnak( 1 ) ] ),
+			null,
+			$guid
+		);
+		$statementTo = new Statement(
+			new PropertyNoValueSnak( 1 ),
+			new SnakList( [ new PropertyNoValueSnak( 2 ) ] ),
+			null,
+			$guid
+		);
+		$latestStatementList = new StatementList( [ $statementLatest ] );
+
+		$differ = new StatementListDiffer();
+		$patcher = new StatementListPatcher();
+
+		$diff = $differ->getDiff( new StatementList( [ $statementFrom ] ), new StatementList( [ $statementTo ] ) );
+		$patcher->patchStatementList( $latestStatementList, $diff );
```

Comment:
This test shows there is an issue either with diffing or with patching. Did you verify the issue is indeed with the latter?

If I wanted to test for the suggested bug I'd have written this test without the differ, and just have a statement and a diff to apply to it. What made you go with the more integration approach?

Reply:
Because I have no clue what Diff should look like to create it by hands.

Reply:
And another reason to have it like this is that `Diff`, as far as I understand, currently is not created anywhere except in the Differ, so we might not treat `Diff`'s constructor as a public interface.
In this case this way of testing it makes perfect sense.

Reply:
+1